### PR TITLE
feat: add kubernetes pss compliance

### LINKED ISSUE #3342: Add Kubernetes PSS compliance
https://kubernetes.io/docs/concepts/security/pod-security-standards

https://kubernetes.slack.com/archives/C8P1DRTJA/p1665615571667579

https://github.com/aquasecurity/trivy/issues/3342#tasklist-block-f2a8eaee-ff89-40f7-9e33-dde5434c90fc

### PR DESCRIPTION
**Pull request by itaysk**
## Description
Depends on updating defsec version, I'm not sure when you want to do that.

## Related issues
- Close #3342 

## Related PRs

Remove this section if you don't have related PRs.

## Checklist
- [ ] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [ ] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/types/report.go [hunk 1]
```diff
@@ -50,11 +50,12 @@ const (
...
 
-	// ComplianceNsa is the compliance checks for nsa
-	ComplianceK8sNsa   = Compliance("k8s-nsa")
-	ComplianceK8sCIS   = Compliance("k8s-cis")
-	ComplianceAWSCIS12 = Compliance("aws-cis-1.2")
-	ComplianceAWSCIS14 = Compliance("aws-cis-1.4")
+	ComplianceK8sNsa             = Compliance("k8s-nsa")
+	ComplianceK8sCIS             = Compliance("k8s-cis")
+	ComplianceK8sPSSBaseline     = Compliance("k8s-pss-baseline")
+	ComplianceK8sPSSRestricted   = Compliance("k8s-pss-restricted")
```
Comment (chen-keinan) [inline pkg/types/report.go:56]: need also to add it to [compliances array ](https://github.com/aquasecurity/trivy/blob/main/pkg/types/report.go#L11)

**Review by chen-keinan — COMMENTED** [review]
also need to update [flag usage](https://github.com/aquasecurity/trivy/blob/main/pkg/commands/app.go#L758)

**Comment by chen-keinan** [comment]
nee also to update [flag usage](https://github.com/aquasecurity/trivy/blob/main/pkg/commands/app.go#L758)

File: pkg/types/report.go (hunk 1 above)
Comment (itaysk) [reply pkg/types/report.go:56]: thanks. fixed now

**Review by chen-keinan — APPROVED** [review]

**Comment by itaysk** [comment]
addressed linter and resolved conflicts

**Comment by knqyf263** [comment]
@chen-keinan Can you review this PR again? Your approval dismissed.

**Comment by chen-keinan** [comment]
@itaysk need to update `go.mod` with latest defsec after merging this [PR](https://github.com/aquasecurity/defsec/pull/1212) and also new `desec` package should be published

**Comment by simar7** [comment]
https://github.com/aquasecurity/defsec/pull/1212 is merged and new defsec with it has been released https://github.com/aquasecurity/defsec/releases/tag/v0.84.0

**Comment by knqyf263** [comment]
@chen-keinan Defsec bumped. Need your review now.

**Review by chen-keinan — APPROVED** [review]

**Comment by knqyf263** [comment]
I found the document was stale. I've updated it in https://github.com/aquasecurity/trivy/pull/3498/commits/5d1a2f8ad3fbcbe05c130b2784e7aefe38e40a9b. Please let me know if I'm mistaken.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**